lib/react.dart

```diff
@@ -8,11 +8,23 @@ library react;
 /// Top-level ReactJS [Component class](https://facebook.github.io/react/docs/top-level-api.html#react.component)
 /// which provides the [ReactJS Component API](https://facebook.github.io/react/docs/component-api.html)
 abstract class Component {
+  Map _props;
+  Map _state = {};
+  dynamic _ref;
+
   /// ReactJS `Component` props.
-  Map props;
+  Map get props => _props;
+  set props(Map value) => _props = value;
+
+  /// ReactJS `Component` state.
+  Map get state => _state;
+  set state(Map value) => _state = value;
 
-  /// Provides access to the underlying DOM representation of the [render]ed `Component`.
-  dynamic ref;
+  /// Returns the component of the specified [ref].
+  ///
+  /// Returns a [Component] if it is a Dart component, otherwise returns an "Dom component" (Dart `Element`).
+  dynamic get ref => _ref;
```

Comment:
While we're in here, can we make the type more specific, as we did in over_react?
```dart
typedef dynamic _RefTypedef(String ref);

_RefTypedef _ref;

_RefTypedef get ref => _ref;
set ref(_RefTypedef  value) => _ref = value;
```